driver/convert.c

```diff
@@ -1605,23 +1599,64 @@ static SQLRETURN parse_iso8601_timestamp(esodbc_stmt_st *stmt, xstr_st *xstr,
 	return SQL_SUCCESS;
 }
 
+BOOL update_dst_date(struct tm *now)
+{
+	int n;
+
+	assert(sizeof(SQLCHAR) == sizeof(char));
+	assert(sizeof(recent_date) - /*\0*/1 == DATE_TEMPLATE_LEN);
+	n = snprintf(recent_date, DATE_TEMPLATE_LEN + 1, "%04d-%02d-%02d",
+			now->tm_year + 1900, now->tm_mon + 1, now->tm_mday);
+	if (n <= 0 || DATE_TEMPLATE_LEN < n) {
+		ERRN("failed to update the DST date.");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static SQLRETURN tss_local_to_utc(esodbc_stmt_st *stmt, TIMESTAMP_STRUCT *tss)
+{
+	struct tm *tmp, tm;
+	time_t utc;
+
+	TIMESTAMP_STRUCT_TO_TM(tss, &tm);
+	tm.tm_isdst = -1;
+	if (((utc = mktime(&tm)) == (time_t)-1) || (! (tmp = gmtime(&utc)))) {
```

Comment:
Similar to `localtime()` above, it would be good to add a `TODO` stating that `gmtime_r()` would need to be used on Linux.